pwiz_tools/Skyline/Model/AbstractModificationMatcher.cs

```diff
@@ -499,7 +499,14 @@ public PeptideDocNode CreateDocNodeFromSettings(LibKey key, Peptide peptide, Srm
             var transitionsUnranked = new List<DocNode>();
             foreach (var peak in spectrum.Peaks)
             {
-                transitionsUnranked.Add(TransitionFromPeakAndAnnotations(key, nodeGroupMatched, fragmentCharge, peak, null));
+                try

```

Comment:
This call should be paying attention to settings and the minimum value that causes the exception reported to initiate this fix. For peptide fragment annotation, we definitely consider the settings, and since we do not rank fragments outside the instrument range. This code also strikes me as odd that you wouldn't just create the precursor and then use a precursor.ChangeSettings(Settings, diff ?? SrmSettingsDiff.ALL) to materialize all of the transitions based on the settings. That way you only write the code once to materialize transitions based on settings.